streams/tck/src/main/java/org/eclipse/microprofile/reactive/streams/tck/OnErrorResumeStageVerification.java

```diff
@@ -0,0 +1,192 @@
+      .map(word -> {
+        if (word.equals("b")) {
+          throw new RuntimeException("failed");
+        }
+        return word.toUpperCase();
+      })
+      .onErrorResumeWith(err -> {
+        exception.set(err);
+        return ReactiveStreams.of("foo", "bar");
+      })
+      .toList()
+      .run(getEngine())), Arrays.asList("A", "foo", "bar"));
+    assertEquals(exception.get().getMessage(), "failed");
+  }
+
+  @Test
+  public void onErrorResumeWithPublisherShouldCatchErrorFromStage() {
+    AtomicReference<Throwable> exception = new AtomicReference<>();
+    assertEquals(await(ReactiveStreams.of("a", "b", "c")
+      .map(word -> {
+        if (word.equals("b")) {
+          throw new RuntimeException("failed");
+        }
+        return word.toUpperCase();
+      })
+      .onErrorResumeWithPublisher(err -> {
+        exception.set(err);
+        return ReactiveStreams.of("foo", "bar").buildRs();
+      })
+      .toList()
+      .run(getEngine())), Arrays.asList("A", "foo", "bar"));
+    assertEquals(exception.get().getMessage(), "failed");
+  }
+
+
+  @Test(expectedExceptions = RuntimeException.class)
+  public void onErrorResumeStageShouldPropagateRuntimeExceptions() {
+    await(ReactiveStreams.failed(new Exception("source-failure"))
+        .onErrorResume(t -> {
+          throw new RuntimeException("failed");
+        })
+        .toList()
+        .run(getEngine()));
+  }
+
+  @Test(expectedExceptions = RuntimeException.class)
+  public void onErrorResumeWithStageShouldPropagateRuntimeExceptions() {
+    await(ReactiveStreams.failed(new Exception("source-failure"))
+      .onErrorResumeWith(t -> {
+        throw new RuntimeException("failed");
+      })
+      .toList()
+      .run(getEngine()));
+  }
+
+  @Test(expectedExceptions = RuntimeException.class)
+  public void onErrorResumeWithPublisherStageShouldPropagateRuntimeExceptions() {
+    await(ReactiveStreams.failed(new Exception("source-failure"))
+      .onErrorResumeWithPublisher(t -> {
+        throw new RuntimeException("failed");
+      })
+      .toList()
+      .run(getEngine()));
+  }
+
+  @Test(expectedExceptions = RuntimeException.class, expectedExceptionsMessageRegExp = ".*boom.*")
+  public void onErrorResumeWithShouldBeAbleToInjectAFailure() {
+    await(ReactiveStreams.failed(new Exception("failed"))
+      .onErrorResumeWith(err -> ReactiveStreams.failed(new Exception("boom")))
+      .toList()
+      .run(getEngine()));
+  }
+
+  @Test(expectedExceptions = RuntimeException.class, expectedExceptionsMessageRegExp = ".*boom.*")
+  public void onErrorResumeWithPublisherShouldBeAbleToInjectAFailure() {
+    await(ReactiveStreams.failed(new Exception("failed"))
+      .onErrorResumeWithPublisher(err -> ReactiveStreams.failed(new Exception("boom")).buildRs())
+      .toList()
+      .run(getEngine()));
+  }
+
+  @Override
+  List<Object> reactiveStreamsTckVerifiers() {
+    return Collections.emptyList();
```

Comment:
Could we return a processor verifier for both resume and resume with that simply rethrew the exception? Also, a resume with publisher verifier that started with a failed publisher, and then resumed with the actual publisher would be good. These will cover all the cases like ensuring that if no exception is thrown, it completes normally, etc.